### ISSUE TITLE
Easy Docker install not working

### ISSUE THREAD

**Issue by gusarg81** [knadh/listmonk#427]
OS: Ubuntu 20.04
Install method: Docker - Production - Easy Docker install

Description: this happens when I exec the one line installer (according to https://listmonk.app/docs/installation/#easy-docker-install_1):

>  sh -c "$(curl -fsSL https://raw.githubusercontent.com/knadh/listmonk/master/install-prod.sh)"
> 
> >  fetching config.toml from listmonk repo
> >  fetching docker-compose.yml from listmonk repo
> >  generating a random password
> >  modifying config.toml
> >  modifying docker-compose.yml
> >  running migrations
> Creating network "mailing_listmonk" with the default driver
> Creating listmonk_db ... done
> >  waiting for db container to be up. retrying in 3s
> >  waiting for db container to be up. retrying in 3s
> >  waiting for db container to be up. retrying in 3s
> >  waiting for db container to be up. retrying in 3s
> >  waiting for db container to be up. retrying in 3s
> Creating mailing_app_run ... done
> 2021/08/05 19:44:05 main.go:84: v1.1.0 (b361292 2021-06-09T14:49:25Z)
> 2021/08/05 19:44:05 init.go:99: reading config: config.toml
> 2021/08/05 19:44:05 init.go:190: connecting to db: listmonk_db:5432/listmonk
> 2021/08/05 19:44:05 init.go:193: error connecting to DB: pq: password authentication failed for user "listmonk"

So, I guess something in wrong with Postgresql User Auth.

Any ideas?

Thanks.

**Comment by mr-karan** [comment]
That's weird. I'll try this on a Ubuntu 20.04 VM and get back. I couldn't reproduce this on my local Fedora box. I suspect anything drastically different on a fresh VM though.

**Comment by gusarg81** [comment]
My case is a Fresh server install as well (Ubuntu server 20.04). I have one container running (Taiga, but even uses a different Postgresql image), so I believe there is no conflict with this docker at all that is causing the problem.

**Comment by mr-karan** [comment]
Tested on a fresh Ubuntu VM:

```
❯ sh -c "$(curl -fsSL https://raw.githubusercontent.com/knadh/listmonk/master/install-prod.sh)"

>  fetching config.toml from listmonk repo
>  fetching docker-compose.yml from listmonk repo
>  generating a random password
>  modifying config.toml
>  modifying docker-compose.yml
>  running migrations
Creating network "testmonk_listmonk" with the default driver
Creating volume "testmonk_listmonk-data" with default driver
Creating listmonk_db ... done
>  waiting for db container to be up. retrying in 3s
>  waiting for db container to be up. retrying in 3s
>  waiting for db container to be up. retrying in 3s
>  waiting for db container to be up. retrying in 3s
>  waiting for db container to be up. retrying in 3s
Creating testmonk_app_run ... done
2021/08/09 03:35:28 main.go:84: v1.1.0 (b361292 2021-06-09T14:49:25Z)
2021/08/09 03:35:28 init.go:99: reading config: config.toml
2021/08/09 03:35:28 init.go:190: connecting to db: listmonk_db:5432/listmonk

** first time installation **
** IMPORTANT: This will wipe existing listmonk tables and types in the DB 'listmonk' **
continue (y/n)?  
```

On comparing the logs, I don't see `Creating volume "testmonk_listmonk-data" with default driver` in your logs. 

I think what has likely happened here is that maybe you ran this script partially on the first run. If yes, then PostgreSQL container would have already initialized with it's own data directory and would not have ran the schema/custom Postgres password.

You can verify it by:

```
docker volume ls | grep listmonk
```

You should see an entry here ^.


To fix this:

```
docker-compose down -v
```

And re-run the script to completion this time. Hopefully that will fix it :)

**Comment by gusarg81** [comment]
Hi,

Ok, now seems to work but conflicts with an existing docker that uses 9000 port. So, should I download the docker.compose.yml manually to modify the port?

Thanks.

**Comment by knadh** [comment]
You can just edit the file.

**Comment by gusarg81** [comment]
Yeah, done after posting. What I did was modifying the install-prod.sh script and change ports modifications there.

```
modify_config(){
    info "generating a random password"
    db_password=$(generate_password)

    info "modifying config.toml"
    # Replace `db.host=localhost` with `db.host=db` in config file.
    sed -i "s/host = \"localhost\"/host = \"listmonk_db\"/g" config.toml
    # Replace `db.password=listmonk` with `db.password={{db_password}}` in config file.
    # Note that `password` is wrapped with `\b`. This ensures that `admin_password` doesn't match this pattern instead.
    sed -i "s/\bpassword\b = \"listmonk\"/password = \"$db_password\"/g" config.toml
    # Replace `app.address=localhost:9000` with `app.address=0.0.0.0:9001` in config file.
    sed -i "s/address = \"localhost:9000\"/address = \"0.0.0.0:9001\"/g" config.toml

    info "modifying docker-compose.yml"
        sed -i "s/POSTGRES_PASSWORD=listmonk/POSTGRES_PASSWORD=$db_password/g" docker-compose.yml
        # Replace port to avoid conflicts with existing Docker container (Taiga in this case)
        sed -i "s/9000:9000/9001:9000/g" docker-compose.yml
}
```

Thanks.

**Comment by gusarg81** [comment]
By the way, there is no connection to the port I've assigned (9001) and the container is running:

```
925f9fec38e5   listmonk/listmonk:latest         "./listmonk"             About an hour ago   Up About an hour             0.0.0.0:9001->9000/tcp, :::9001->9000/tcp                              listmonk_app
8d57fcc80faf   postgres:11                      "docker-entrypoint.s…"   About an hour ago   Up About an hour (healthy)   0.0.0.0:9432->5432/tcp, :::9432->5432/tcp                              listmonk_db
```

Any ideas?

**Comment by MRobi1** [comment]
> By the way, there is no connection to the port I've assigned (9001) and the container is running:
> 
> ```
> 925f9fec38e5   listmonk/listmonk:latest         "./listmonk"             About an hour ago   Up About an hour             0.0.0.0:9001->9000/tcp, :::9001->9000/tcp                              listmonk_app
> 8d57fcc80faf   postgres:11                      "docker-entrypoint.s…"   About an hour ago   Up About an hour (healthy)   0.0.0.0:9432->5432/tcp, :::9432->5432/tcp                              listmonk_db
> ```
> 
> Any ideas?
> 
> EDIT: Well, changed to 0.0.0.0:9001->9001 and now works.

@gusarg81 I'm wondering if you could elaborate on this a little?
I used your modification above and the container deploys fine but can't connect through port 9001.
Tried changing "0.0.0.0:9001\" to just "9001\" as implied above and can't connect. Tried changing to the IP of the device and can't connect.

TONS of people run portainer which runs on port 9000, you'd think a different port would be the default.

**Comment by afonsolopez** [comment]
> Yeah, done after posting. What I did was modifying the install-prod.sh script and change ports modifications there.
> 
> ```
> modify_config(){
>     info "generating a random password"
>     db_password=$(generate_password)
> 
>     info "modifying config.toml"
>     # Replace `db.host=localhost` with `db.host=db` in config file.
>     sed -i "s/host = \"localhost\"/host = \"listmonk_db\"/g" config.toml
>     # Replace `db.password=listmonk` with `db.password={{db_password}}` in config file.
>     # Note that `password` is wrapped with `\b`. This ensures that `admin_password` doesn't match this pattern instead.
>     sed -i "s/\bpassword\b = \"listmonk\"/password = \"$db_password\"/g" config.toml
>     # Replace `app.address=localhost:9000` with `app.address=0.0.0.0:9001` in config file.
>     sed -i "s/address = \"localhost:9000\"/address = \"0.0.0.0:9001\"/g" config.toml
> 
>     info "modifying docker-compose.yml"
>         sed -i "s/POSTGRES_PASSWORD=listmonk/POSTGRES_PASSWORD=$db_password/g" docker-compose.yml
>         # Replace port to avoid conflicts with existing Docker container (Taiga in this case)
>         sed -i "s/9000:9000/9001:9000/g" docker-compose.yml
> }
> ```
> 
> Thanks.

You saved me man! Awesome!
